rust-runtime/aws-smithy-http-server-python/examples/pokemon-service-test/tests/helpers.rs

```diff
@@ -3,9 +3,61 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+use std::fs::File;
+use std::io::BufReader;
+use std::process::Command;
+use std::time::Duration;
+
+use aws_smithy_client::{erase::DynConnector, hyper_ext::Adapter};
+use aws_smithy_http::operation::Request;
 use command_group::{CommandGroup, GroupChild};
 use pokemon_service_client::{Builder, Client, Config};
-use std::{process::Command, thread, time::Duration};
+use tokio::time;
+
+const TEST_KEY: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/tests/testdata/localhost.key");
+const TEST_CERT: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/tests/testdata/localhost.crt");
+
+pub type PokemonClient = Client<
+    aws_smithy_client::erase::DynConnector,
+    aws_smithy_client::erase::DynMiddleware<aws_smithy_client::erase::DynConnector>,
+>;
+
+enum PokemonServiceVariant {
+    Http,
+    Http2,
+}
+
+impl PokemonServiceVariant {
+    async fn run_process(&self) -> GroupChild {
+        let mut args = vec!["../pokemon_service.py".to_string()];
+
+        match self {
+            PokemonServiceVariant::Http => {}
+            PokemonServiceVariant::Http2 => {
+                args.push("--enable-tls".to_string());
+                args.push(format!("--tls-key-path={TEST_KEY}"));
+                args.push(format!("--tls-cert-path={TEST_CERT}"));
+            }
+        }
+
+        let process = Command::new("python3")
+            .args(args)
+            .group_spawn()
+            .expect("failed to spawn the Pokémon Service program");
+
+        // The Python interpreter takes a little to startup.
+        time::sleep(Duration::from_secs(2)).await;
```

Comment:
This is not racy since it is in a test and it has been working both in CI and locally for months. Python takes a while to startup, that's it. The effort of using a signal would be useless IMHO since this works perfectly already. We do the same in Rust BTW, just a smaller sleep because its faster to startup.